File: backend/src/main/java/com/mongilbasket/group/Group.java
package com.mongilbasket.group;

import java.time.LocalTime;
import java.util.UUID;

import com.mongilbasket.coach.Coach;
import com.mongilbasket.season.Season;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "groups")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Group {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "season_id", nullable = false)
    private Season season;

    @Column(nullable = false)
    private int ageMin;

    @Column(nullable = false)
    private int ageMax;

    @Column(nullable = false)
    private int capacity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "coach_id")
    private Coach coach;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ScheduleDay scheduleDay;

    @Column(nullable = false)
    private LocalTime scheduleStartTime;

    @Column(nullable = false)
    private LocalTime scheduleEndTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private GroupStatus status = GroupStatus.ACTIVE;
}

File: mobile/lib/features/groups/group.dart
class Group {
  const Group({
    required this.id,
    required this.name,
    required this.seasonId,
    required this.ageMin,
    required this.ageMax,
    required this.capacity,
    required this.currentCount,
    required this.status,
    this.coachName,
  });

  factory Group.fromJson(Map<String, dynamic> json) => Group(
        id: json['id'] as String,
        name: json['name'] as String,
        seasonId: json['seasonId'] as String,
        ageMin: json['ageMin'] as int,
        ageMax: json['ageMax'] as int,
        capacity: json['capacity'] as int,
        currentCount: json['currentCount'] as int,
        coachName: json['coachName'] as String?,
        status: json['status'] as String,
      );

  final String id;
  final String name;
  final String seasonId;
  final int ageMin;
  final int ageMax;
  final int capacity;
  final int currentCount;
  final String? coachName;
  final String status;

  bool get isFull => currentCount >= capacity;
}
